mlir/lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -964,6 +975,8 @@ static LogicalResult printOperation(CppEmitter &emitter, emitc::ForOp forOp) {
 
   os.unindent() << "}";
 
+  emitter.decreaseLoopNestingLevel();
```

Comment:
Alternative Idea: Create a CppEmitter::LoopScope that inherits from CppEmitter:Scope, and make it call `decreaseLoopNestingLevel` in its destructore? It's too easy to early return from this function and forget the call.